nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,631 @@
+"""High-level decoding object that exposes standard classification
+and regression strategies such as SVM, LogisticRegression and Ridge,
+with optional feature selection, and integrated parameter selection.
+"""
+# Authors : Yannick Schwartz
+#           Andres Hoyos-Idrobo
+#
+# License: simplified BSD
+
+import collections
+import itertools
+import warnings
+import copy
+
+import numpy as np
+
+from sklearn.externals.joblib import Parallel, delayed, Memory
+from sklearn.linear_model.ridge import Ridge, RidgeClassifier, _BaseRidge
+from sklearn.linear_model import LogisticRegression
+from sklearn.svm import LinearSVC, SVR
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.feature_selection import SelectPercentile, SelectKBest
+from sklearn.feature_selection import f_classif, f_regression
+from sklearn.svm.bounds import l1_min_c
+from sklearn.metrics import make_scorer
+from sklearn.metrics import r2_score, f1_score, precision_score, recall_score
+from sklearn.metrics.scorer import check_scoring
+from sklearn.grid_search import ParameterGrid
+from sklearn.base import BaseEstimator
+from sklearn.base import is_classifier
+from sklearn.utils import check_X_y, check_consistent_length
+from sklearn import clone
+
+try:
+    from sklearn.utils import check_arrays as check_array
+except ImportError as e:
+    from sklearn.utils import check_array
+
+try:
+    # scikit-learn < 0.17
+    from sklearn.cross_validation import _check_cv as check_cv
+except ImportError as e:
+    # scikit-learn >= 0.17
+    from sklearn.cross_validation import check_cv
+
+from ..input_data import NiftiMasker, MultiNiftiMasker
+from ..image import index_img
+
+
+# Volume of a standard (MNI152) brain mask in mm^3
+MNI152_BRAIN_VOLUME = 1827243.
+
+
+ESTIMATOR_CATALOG = dict(
+    svc_l1=LinearSVC(penalty='l1', dual=False, class_weight='auto'),
+    svc_l2=LinearSVC(penalty='l2', class_weight='auto'),
+    svc=LinearSVC(penalty='l2', class_weight='auto'),
+    logistic_l1=LogisticRegression(penalty='l1', class_weight='auto'),
+    logistic_l2=LogisticRegression(penalty='l2', class_weight='auto'),
+    logistic=LogisticRegression(penalty='l2', class_weight='auto'),
+    ridge_classifier=RidgeClassifier(class_weight='auto'),
+    ridge_regression=Ridge(),
```

Comment:
It is possible to use other estimators (at your own risk), for instance 
Decoder(estimator = Lasso(alpha=1.))